### ISSUE TITLE
When using the player to play postrolls, the video always replays after postrolls

### ISSUE THREAD

**Issue by apesant** [videojs/videojs-contrib-ads#160]
When using the following code : 

```
function contentEnded(){
            player.ads.startLinearAdMode();
            player.src(""); // In my use case I have "real" postrolls to play
            setTimeout(player.ads.endLinearAdMode, 5000);
        }

player.on('contentended', contentEnded);
```

the video restarts after the postrolls.

> VIDEOJS: ads ended triggered: content-resuming -> content-playback
> video.js:19842 VIDEOJS: ads contentended triggered: content-playback -> postroll?
> video.js:19842 VIDEOJS: ads adstart triggered: postroll? -> ad-playback
> video.js:19842 VIDEOJS: ERROR: (CODE:4 MEDIA_ERR_SRC_NOT_SUPPORTED) The media could not be loaded, either because the server or network failed or because the format is not supported. MediaError {code: 4, message: "The media could not be loaded, either because the …rk failed or because the format is not supported."}_logType @ video.js:19842log.error @ video.js:19793error @ video.js:12124handleTechError_ @ video.js:10929ret @ video.js:19671ret @ video.js:19671data.dispatcher @ video.js:19319
> video.js:19842 VIDEOJS: ads adend triggered: ad-playback -> content-resuming
> videojs.pulse.js:124 Object {type: "contentplayback", triggerevent: "ended", target: document, relatedTarget: undefined, defaultPrevented: false…}
> video.js:19842 VIDEOJS: ads ended triggered: content-resuming -> content-playback

Am I doing something wrong?

I am using the latest versions of the plugin and videojs.

**Comment by incompl** [comment]
I want to understand what's going on to help solve your issue. I just spent some time trying to reproduce this in 2 ad integrations and the example integration and I haven't been able to. 

I noticed in the [example integration](https://github.com/videojs/videojs-contrib-ads/blob/master/example/example-integration.js) `endLinearAdMode` happens on `adended`, not on `contentended`. Maybe that's related to your issue? Are you able to reproduce your issue in the example integration, or modify the example integration to demonstrate your issue? I think that would help pinpoint what's going on.

**Comment by apesant** [comment]
Thanks for your reply!

I can not use adEnded with our SDK because we actually only use the HTML5 video element of videojs to play ads (without using the player API). So I get an event AD_FINISHED from our SDK when an ad is finished.

It is the exact same use case than Google IMA's integration, which also has to deal with that problem it seems : https://github.com/googleads/videojs-ima/blob/master/src/videojs.ima.js#L354

I do the same in my code, I don't use the endLinearAdMode() API after the postrolls to avoid the video to autoreplay.

Is this use case (when the video element is used by the ad plugin directly to play the ads) something you could consider in the contrib-ads plugin?

**Comment by incompl** [comment]
I believe this was fixed in version 3.3.10. Please reopen or create a new issue if seen again.
